pkg/plugins/golang/declarative/v1/scaffolds/api.go

```diff
@@ -70,8 +71,15 @@ func (s *apiScaffolder) Scaffold() error {
 		machinery.WithResource(&s.resource),
 	)
 
+	isLegacyLayout := false
+	for _, pluginKey := range s.config.GetPluginChain() {
+		if strings.Contains(pluginKey, "go.kubebuilder.io/v3") {
+			isLegacyLayout = true
+		}
+	}
```

Comment:
This code is duplicated. Would it be possible to avoid it by adding a `LegacyLayout()` func on the `Config` struct?

Reply:
The Config structure is what we have in the Project file. 
This code checks that the project was scaffolded with the previous plugin versions.
Therefore, we use the info to know if is or not a legacylayout (projects scaffold within versions go/v2 (will be removed in the next+1 release) or go/v3 - v1 is no longer supported by the tool/removed ).